subsys/bluetooth/services/dis.c

```diff
@@ -64,13 +77,32 @@ static uint8_t dis_hw_rev[CONFIG_BT_DIS_STR_MAX] =
 static uint8_t dis_sw_rev[CONFIG_BT_DIS_STR_MAX] =
 	CONFIG_BT_DIS_SW_REV_STR;
 #endif
+#if defined(CONFIG_BT_DIS_UDI)
+static uint8_t dis_udi_label[CONFIG_BT_DIS_STR_MAX] =
+	CONFIG_BT_DIS_UDI_LABEL_STR;
```

Comment:
If `CONFIG_BT_DIS_STR_MAX` is smaller than any of the provided strings, does this throw a compile error?

Reply:
No, it does not. It gives a warning:
```
warning: initializer-string for array of 'unsigned char' is too long
```

The effect is that the string literal is truncated, so the null terminator is not included.
This is a bug in the existing implementation.

Reply:
Additionally, the Kconfig help for `BT_DIS_STR_MAX` says:
```Bluetooth DIS string storage size. Storage can be up to 248 bytes
	  long (excluding NULL termination)
```

It sounds as if the string literals can contains 248 characters, maybe the arrays should be `BT_DIS_STR_MAX + 1` long.

Reply:
> It sounds as if the string literals can contains 248 characters, maybe the arrays should be BT_DIS_STR_MAX + 1 long.

That won't help the issue if they are still just truncated without the NULL terminator. 
So the arrays need to be `BT_DIS_STR_MAX + 1` **and** you need to ensure that it's NULL terminated at the time of read

Reply:
It does not seem like we can give compiler errors, since `sizeof` cannot be evaluated by the preprocessor. But if we could live with the warning and then just set a null-terminator manually, that could work.

Reply:
We don't actually have to set a null-terminator. We could simply swap `strlen` in the read method for `strnlen` when settings is defined. Then we do not need the null-terminator, so we could store a string of CONFIG_BT_DIS_STR_MAX characters in an array of that size, so no need to add 1 to the size.

Reply:
> It does not seem like we can give compiler errors, since `sizeof` cannot be evaluated by the preprocessor.

`BUILD_ASSERT(sizeof(...) < ...)` works.

Reply:
I have added `BUILD_ASSERT` and removed `BT_DIS_STRLEN`.